call_test.go

```diff
@@ -0,0 +1,215 @@
+package yarpc
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/api/transport/transporttest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOutboundCallWriteToRequest(t *testing.T) {
+	tests := []struct {
+		desc        string
+		giveOptions []CallOption
+		giveRequest transport.Request
+		wantRequest transport.Request
+	}{
+		{
+			desc:        "no options",
+			giveOptions: []CallOption{},
+			giveRequest: transport.Request{
+				Caller:    "caller",
+				Service:   "service",
+				Encoding:  transport.Encoding("raw"),
+				Procedure: "hello",
+			},
+			wantRequest: transport.Request{
+				Caller:    "caller",
+				Service:   "service",
+				Encoding:  transport.Encoding("raw"),
+				Procedure: "hello",
+			},
+		},
+		{
+			desc: "headers",
+			giveOptions: []CallOption{
+				WithHeader("foo", "bar"),
+				WithHeader("baz", "qux"),
+			},
+			giveRequest: transport.Request{
+				Caller:    "caller",
+				Service:   "service",
+				Encoding:  transport.Encoding("raw"),
+				Procedure: "hello",
+			},
+			wantRequest: transport.Request{
+				Caller:    "caller",
+				Service:   "service",
+				Encoding:  transport.Encoding("raw"),
+				Procedure: "hello",
+				Headers:   transport.HeadersFromMap(map[string]string{"foo": "bar", "baz": "qux"}),
```

Comment:
No, we don't sort them. The placeholders are usually used in the progression foo, bar, baz, qux, quux.